### PR TITLE
cloudinit configs, private keys via AWS Secrets Manager, Policies for same, other tidying and updating

### PR DESCRIPTION
**Pull request by benjaminsherwood**
Signed-off-by: Benjamin Sherwood <benjamin.sherwood@ext.ons.gov.uk>

### CODE REVIEW

File: concourse_web.tf [hunk 1]
```diff
@@ -3,10 +3,9 @@ resource "aws_launch_template" "concourse_web" {
   image_id                             = var.ami_id
   instance_type                        = var.concourse_web_conf.instance_type
   instance_initiated_shutdown_behavior = "terminate"
+  key_name                             = "concourse"
```
Comment (dan-hill2802) [inline concourse_web.tf:6]: can you make this a variable and optional?

File: concourse_web.tf [hunk 2]
```diff
@@ -49,7 +48,8 @@ resource "aws_launch_template" "concourse_web" {
     delete_on_termination       = true
 
     security_groups = [
-      aws_security_group.concourse_web.id
+      aws_security_group.concourse_web.id,
+      aws_security_group.concourse_vpc_endpoints.id
```
Comment (dan-hill2802) [inline concourse_web.tf:52]:
```suggestion
      aws_security_group.concourse_vpc_endpoints.id,
```

File: diagram.drawio [hunk 3]
```diff
@@ -1 +1 @@
...
.../O3oX6t1+0Qi7JqTaRSSqBi1KmYgK5d68t+5dqULZimkv46XLJmyRR6iV7fca5YWomasmHttKt65u6lJFTa8c137kaqPXUbYZs+7RZBLHNa8sf/h+6+6c+lD9fVKI1Dv96ajao9v/AEG63tj3XccHIkT/GqxcrbEp2o/SVENKuo1oivTTbcIg8uYzWbdxZl40nXfUbSxPllyl055Tv8b2qJ5TCaUjgTofpAM5OzuQbY7JGr8qY+mqA9ktSFknHWg+j5yZtAOFrj+tMQSO70COKzEEuuw/tmzRYENcR5v+Uw2vRkFX6mehf4r+5OpjTxyQuutPhkTcP/GA5Dljo+cOVb8i9TIgddCBfG3s9DkgDWUIhXbkhZasA3nG1HScqq0qn3o5pDfZg5pFdkdmEQ543TEi/dwdCshxfxvJ7KxHDWUj0QxF9VmOuu9R/VtMtjjToqRTVRnfBxuhDG1nh4I2k+FXfkSBu131LkckfDVrfDmhHJWeB7o3bcsFxvn22qTSVVcwx5nYI+udfXP0c1Xjjx0yiY80YQ9Zl4mzd0OQyvJ4z7aCeFxiMTyaUkzun9nHdS0Gg4bpt7QsybnjYCeyYLo+mR936XLlye8wgdm/W2+7vAcafcue3Bv7odG+NrUbaQjFT4lGtHru0VYHQJtVgpItlDqDnyt6NT5i8PX+UiOxs8Qd1WMEiCsLLxTS/VWWso6iRcayjH3SFkLSMSqbE3IatKahZnXR7hxRdWut+C+QFSyCZf/pr9112J8/SacyJVtBtOdZsq0t2LSlOBU+l3NeGS65tJ26JtuRTWajKMGmJ7qhq8mBVA5WnWU2pKKslTG3vGe41IYJV/QMi1MJknTdgCkLtykPPQZYJ5zyULIxomgpXwY1ichNF4hM2/6wzMQYe6IDsathTr4dhGSvto+yiKVpb62GccFmuy/dOnMwUeqiz/KjLG05TpQuSQdPDAa7v5DIGlF+3KVlakVJV00MJ0rR6flRVt8cJ0oLjJ1kdCJGYY9LcmukKZqKH2VZzpHDpU+9gHSyZ2gta4i21UdZRHCkMD02W7ulW0OLUnS7fZRFPMeJkiaAJ71ycBUrWex76ZUtRKmbFqdh9R6X0dXIUnQcXLplK/JjOzz50cfViVzf63EBvly4pjh8XrhQO+HyXMjtzEq5KtM0ba9VEjSZd/8P</diagram></mxfile>
```
Comment (dan-hill2802) [inline diagram.drawio:1]: I prefer editable PNGs, but :man_shrugging:

File: iam.tf [hunk 4]
```diff
@@ -49,6 +49,58 @@ resource "aws_iam_role_policy_attachment" "concourse_worker_ssm" {
...
+      "secretsmanager:GetSecretValue"
+    ]
+
+    resources = [
+      var.concourse_sec.session_signing_key_private_secret_arn,
+      var.concourse_sec.session_signing_key_public_secret_arn,
+      var.concourse_sec.tsa_host_key_private_secret_arn,
+      var.concourse_sec.tsa_host_key_public_secret_arn,
+      var.concourse_sec.worker_key_private_secret_arn,
+      var.concourse_sec.worker_key_public_secret_arn
```
Comment (dan-hill2802) [inline iam.tf:75]:
```suggestion
      var.concourse_sec.worker_key_public_secret_arn,
```

File: iam.tf [hunk 5]
```diff
@@ -49,6 +49,58 @@ resource "aws_iam_role_policy_attachment" "concourse_worker_ssm" {
...
+
+data "aws_iam_policy_document" "concourse_worker_secrets_manager" {
+  statement {
+    actions = [
+      "secretsmanager:GetSecretValue"
+    ]
+
+    resources = [
+      var.concourse_sec.tsa_host_key_public_secret_arn,
+      var.concourse_sec.worker_key_private_secret_arn
```
Comment (dan-hill2802) [inline iam.tf:99]:
```suggestion
      var.concourse_sec.worker_key_private_secret_arn,
```

**Review by dan-hill2802 — COMMENTED** [review]
with this still being quite WIP, I've done more of a glance than a review

File: concourse_web.tf (hunk 1 above)
Comment (benjaminsherwood) [reply concourse_web.tf:6]: nope, but i can remove it as it's an oversight from the SSM troubleshooting, and is no longer required :)

File: diagram.drawio (hunk 3 above)
Comment (benjaminsherwood) [reply diagram.drawio:1]: Yea, this needs updating to a PNG file, just not bothered yet.

**Review by dan-hill2802 — APPROVED** [review]
caveat that this is WIP

**Pull request merged**